modules/patient-portal/src/main/java/io/uhndata/cards/patients/internal/DraftsAnswersCleanupTask.java

```diff
@@ -0,0 +1,161 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.uhndata.cards.patients.internal;
+
+import java.time.ZonedDateTime;
+import java.util.Iterator;
+import java.util.Map;
+
+import javax.jcr.Node;
+import javax.jcr.NodeIterator;
+import javax.jcr.RepositoryException;
+import javax.jcr.query.Query;
+
+import org.apache.sling.api.resource.LoginException;
+import org.apache.sling.api.resource.PersistenceException;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ResourceResolverFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.uhndata.cards.patients.api.PatientAccessConfiguration;
+import io.uhndata.cards.resolverProvider.ThreadResourceResolverProvider;
+
+/**
+ * Periodically remove patient's answer values with a frequency specified by @{code PatientAccess.draftLifetime}
+ * that haven't been submitted by the patient.
+ *
+ * @version $Id$
+ * @since 0.9.6
+ */
+public class DraftsAnswersCleanupTask implements Runnable
+{
+
+    /** Default log. */
+    private static final Logger LOGGER = LoggerFactory.getLogger(DraftsAnswersCleanupTask.class);
+
+    /** Provides access to resources. */
+    private final ResourceResolverFactory resolverFactory;
+
+    /** For sharing the resource resolver with other services. */
+    private final ThreadResourceResolverProvider rrp;
+
+    /** Grab details on the number of days draft responses from patients are kept. */
+    private final PatientAccessConfiguration patientAccessConfiguration;
+
+    /**
+     * @param resolverFactory a valid ResourceResolverFactory providing access to resources
+     * @param rrp sharing the resource resolver with other services
+     * @param patientAccessConfiguration details on the number of days draft responses from patients are kept
+     */
+    DraftsAnswersCleanupTask(final ResourceResolverFactory resolverFactory, final ThreadResourceResolverProvider rrp,
+        final PatientAccessConfiguration patientAccessConfiguration)
+    {
+        this.resolverFactory = resolverFactory;
+        this.rrp = rrp;
+        this.patientAccessConfiguration = patientAccessConfiguration;
+    }
+
+    @Override
+    public void run()
+    {
+        boolean mustPopResolver = false;
+        try (ResourceResolver resolver = this.resolverFactory
+            .getServiceResourceResolver(Map.of(ResourceResolverFactory.SUBSERVICE, "VisitFormsPreparation"))) {
+            this.rrp.push(resolver);
+            mustPopResolver = true;
+            final int draftLifetime = this.patientAccessConfiguration.getDraftLifetime();
+            // If draftLifetime is -1, do nothing
+            // (-1 means never delete; any cleanup that is necessary will be done by UnsubmittedFormsCleanupTask)
+            if (draftLifetime == -1) {
```

Comment:
Technically it is possible to get a lower negative value, which might get us in trouble at line 110. We should either reject any negative value here with 
```suggestion
            if (draftLifetime <= -1) {
```
or, better, replace any negative value coming from the node with `-1` in [PatientAccessConfigurationImpl.java#getDraftLifetime](https://github.com/data-team-uhn/cards/blob/dev/modules/patient-portal/src/main/java/io/uhndata/cards/patients/internal/PatientAccessConfigurationImpl.java#L141).

Reply:
Addressed in [c3297d2](https://github.com/data-team-uhn/cards/pull/1298/commits/c3297d2b193e815eb298777d09ac793ca9ae236b)